data/helpers.d/system

```diff
@@ -53,3 +53,92 @@ ynh_abort_if_errors () {
 ynh_get_debian_release () {
 	echo $(lsb_release --codename --short)
 }
+
+# Start (or other actions) a service,  print a log in case of failure and optionnaly wait until the service is completely started
+#
+# usage: ynh_systemd_action [-n service_name] [-a action] [ [-l "line to match"] [-p log_path] [-t timeout] [-e length] ]
+# | arg: -n, --service_name= - Name of the service to reload. Default : $app
+# | arg: -a, --action=       - Action to perform with systemctl. Default: start
+# | arg: -l, --line_match=   - Line to match - The line to find in the log to attest the service have finished to boot.
+#                              If not defined it don't wait until the service is completely started.
+# | arg: -p, --log_path=     - Log file - Path to the log file. Default : /var/log/$app/$app.log
+# | arg: -t, --timeout=      - Timeout - The maximum time to wait before ending the watching. Default : 300 seconds.
+# | arg: -e, --length=       - Length of the error log : Default : 20
+ynh_systemd_action() {
+    # Declare an array to define the options of this helper.
+    declare -Ar args_array=( [n]=service_name= [a]=action= [l]=line_match= [p]=log_path= [t]=timeout= [e]=length= )
+    local service_name
+    local action
+    local line_match
+    local length
+    local log_path
+    local timeout
+
+    # Manage arguments with getopts
+    ynh_handle_getopts_args "$@"
+
+    local service_name="${service_name:-$app}"
+    local action=${action:-start}
+    local log_path="${log_path:-/var/log/$service_name/$service_name.log}"
+    local length=${length:-20}
+    local timeout=${timeout:-300}
+
+    # Start to read the log
+    if [[ -n "${line_match:-}" ]]
+    then
+        local templog="$(mktemp)"
+	# Following the starting of the app in its log
+	if [ "$log_path" == "systemd" ] ; then
+            # Read the systemd journal
+            journalctl -u $service_name -f --since=-45 > "$templog" &
+	else
+            # Read the specified log file
+            tail -F -n0 "$log_path" > "$templog" &
```

Comment:
Because that's exactly the point, we don't want to find a previous line.
We're going to restart the service [here](https://github.com/YunoHost/yunohost/pull/633/files#diff-43d93913a4448bf0d9e96b5bc480bf5aR103), so first we start to tail the log, then we restart the service.